src/mfs.jl

```diff
@@ -0,0 +1,163 @@
+
+""" 
+    check_basin(mapper, X, Y)
+
+Input:
+    mapper: AttractorMapper
+    X: Point 1
+    Y: Point 2
+Output: 
+    true if X and Y are in the same basin of attraction, false otherwise
+
+Description:
+    This function checks if two points are in the same basin of attraction. 
+    It does so by checking if the points are mapped to the same point by the mapper.
+
+"""
+
+function check_basin(mapper::AttractorMapper, X, Y)
+    idx = mapper(X)
+    idy = mapper(Y)
+    if idx == idy
+        return true
+    else
+        return false
+    end
+end
```

Comment:
This function needs to be removed. First, there is no reason to dedicate a new function and 8 lines of code to what is literally the operation `mapper(X) == mapper(Y)`. But most importantly, `Y` here is always the same, it is the starting initial condition. There is no reason to calculate its mapped id more than once, so do that and save the mapped id and then just compare `mapper(X) == id_u0`.